openmdao/solvers/linear/linear_schur.py

```diff
@@ -0,0 +1,381 @@
+                             f'{len(self._sys_names)}.\n{self._sys_names}')
+
+    def solve(self, mode, rel_systems=None):
+        """
+        Perform the operations in the iteration loop.
+
+        Parameters
+        ----------
+        mode : str
+            Specifies whether the solver is operating in forward ('fwd') or reverse ('rev') mode.
+        rel_systems : set of str
+            Set of names of relevant systems based on the current linear solve.  Deprecated.
+        """
+        self._mode = mode
+        self._rel_systems = rel_systems
+        system = self._system()
+        mode = self._mode_linear
+
+        if self._mode == 'fwd':
+            self._rhs_vec = system._dresiduals.asarray(True)
+        else:
+            self._rhs_vec = system._doutputs.asarray(True)
+
+        # take the subsystems
+        subsys1, _ = system._subsystems_allprocs[self._sys_names[0]]
+        subsys2, _ = system._subsystems_allprocs[self._sys_names[1]]
+
+        # TODO this may not be the most general case. think about just solving for a subset
+        subsys2_outputs = subsys2._doutputs
+        subsys2_residuals = subsys2._dresiduals
+
+        # list of variables we solve for here. this should include all variables in
+        # subsys2 ideally because we dont do anything else for this subsystem here.
+        vars_to_solve = [*subsys2_outputs.keys()]
+        resd_to_solve = [*subsys2_residuals.keys()]
+
+        # total size of the jacobian
+        n_vars = 0
+        for var in vars_to_solve:
+            n_vars += subsys2_outputs[var].size
+
+        # initialize the schur complement jacobian for these variables
+        # TODO better way to get the dtype?
+        schur_jac = np.zeros((n_vars, n_vars),
+                             dtype=system._vectors['residual']['linear'].asarray(copy=True).dtype)
+        schur_rhs = np.zeros((n_vars),
+                             dtype=system._vectors['residual']['linear'].asarray(copy=True).dtype)
+
+        # backup the vectors we are working with
+        rvec = system._vectors['residual']['linear']
+        ovec = system._vectors['output']['linear']
+        ivec = system._vectors['input']['linear']
+
+        if mode == 'fwd':
+            parent_offset = system._dresiduals._root_offset
+
+            # take the d_resdiuals for both of the subsys
+            b_vec = subsys1._dresiduals
+            off = b_vec._root_offset - parent_offset
+            b_vec2 = subsys2._dresiduals
+            off2 = b_vec2._root_offset - parent_offset
+
+            # cache the rhs vector since we ll need this later
+            subsys1_rhs = self._rhs_vec[off: off + len(b_vec)].copy()
+            subsys2_rhs = self._rhs_vec[off2: off2 + len(b_vec2)].copy()
+
+            # Beg solve for subsys 2
+            # schur_jacobian
+
+            # Schur_Jac = D - C A^-1 B ##
+
+            # put back the vectors
+            rvec.set_val(np.zeros(len(rvec)))
```

Comment:
You don't need to create temporary np.zeros arrays here.  You can just do, e.g., rvec.set_val(0.0) and that will zero out the whole vector.